### PR TITLE
Clustering Policy Update Implementation

### PR DESCRIPTION
**Pull request by JackKuei**
Prior to a PR being reviewed, there needs to be a Github issue that the PR
addresses. Replace the brackets and text below with that issue number.

For #[823 [Add support for Senlin clustering]](https://github.com/gophercloud/gophercloud/issues/823)

Links to the line numbers/files in the OpenStack source code that support the
code in this PR:

[[policies:update]](https://github.com/openstack/senlin/blob/e5bc16b3fea2f10dead365f983c5a937f23d72b0/senlin/api/openstack/v1/policies.py#L74)

### CODE REVIEW

**Comment by JackKuei** [comment]
recheck

**Comment by jtopjian** [comment]
@JackKuei This will need rebased from the Validate merge.

File: openstack/clustering/v1/policies/requests.go [hunk 1]
```diff
@@ -86,13 +86,52 @@ func Create(client *gophercloud.ServiceClient, opts CreateOpts) (r CreateResult)
...
+}
+
+// UpdateOptsBuilder builder
+type UpdateOptsBuilder interface {
+	ToPolicyUpdateMap() (map[string]interface{}, error)
+}
+
+// UpdateOpts params
+type UpdateOpts struct {
+	Policy map[string]interface{} `json:"-"`
```
Comment (jtopjian) [inline openstack/clustering/v1/policies/requests.go]: I don't think `Policy` is a valid field. Looking at the Senlin code, a Policy Update request is changed into a `PolicyUpdateRequestBody` object which only has a "name" field:

https://github.com/openstack/senlin/blob/a0f875b711d794629581c9643a8bc9594d1db4b3/senlin/objects/requests/policies.py#L58

In the API docs where it has "policy" as an object, I think it means the JSON body:

```json
"policy": {
  "name": "new-name"
}
```

which is taken care of on line 114.

According to this, too, "name" is the only field able to be updated: https://github.com/openstack/senlin/blob/a7fea71b9ee6de096349e1322400dbf84c075634/senlin/engine/service.py#L627-L638

It looks like the full policy (with the new name) is returned, though, so no changes needed there.

**Comment by jtopjian** [comment]
LGTM - thank you :)

**Pull request merged**